src/pyeval.jl

```diff
@@ -202,32 +202,37 @@ pasted into the Python code.   This allows you to evaluate code
 where the code itself is generated by a Julia expression.
 """
 macro py_str(code, options...)
+    m = :(pynamespace($__module__))
+    esc(macroexpand(__module__, :(PyCall.@_py_str($m, $m, $code, $(options...)))))
+end
+
+macro _py_str(pyglobals, pylocals, code, options...)
     T = length(options) == 1 && 'o' in options[1] ? PyObject : PyAny
     code, locals = interpolate_pycode(code)
     input_type = '\n' in code ? Py_file_input : Py_eval_input
     fname = make_fname(@__FILE__)
```

Comment:
Should it use `__source__.file`, BTW?

Reply:
Yea, I think so. Before I think it was broken since `@__FILE__` would always just return `pycall.jl`. I also think the the `make_fname` stuff is unnecessary, `__source__.file` never seems to return `nothing`. Will add this change.